pkg/controllers/streaming/processor_controller.go

```diff
@@ -428,7 +428,9 @@ func (r *ProcessorReconciler) constructDeploymentForProcessor(processor *streami
 
 	volumes := []corev1.Volume{}
 	volumeMounts := []corev1.VolumeMount{}
-	for _, stream := range streams {
+	streams := append(inputStreams, outputStreams...)
```

Comment:
This can modify the underlying array of `inputStreams` as a side effect. While this is harmless here, I think it may become problematic if `inputStreams` is later re-used and could lead to subtle bug.
Same goes for the line below.

Reply:
done